### ISSUE TITLE
SyncPlay broken since 20200703.1-unstable

### ISSUE THREAD

**Issue by Spiritreader** [jellyfin/jellyfin#3494]
**Describe the bug**
SyncPlay no longer functions. I am pretty certain that this is related to:
- #3378 
- https://github.com/jellyfin/jellyfin-web/pull/1475
- jellyfin/jellyfin-apiclient-javascript#54

**System (please complete the following information):**
 - OS: Unraid 6.8.3
 - Virtualization: Docker
 - Jellyfin Version: 20200703.1-unstable
 - Reverse Proxy: with and without reverse proxy
 - Base URL: none
 - Networking: All networking modes

**To Reproduce**

1. Click on "SyncPlay"

**Expected behavior**
Sync Play shows groups

**Logs**
```
[2020-07-03 20:02:12.451 +02:00] [ERR] [65] Emby.Server.Implementations.HttpServer.HttpListenerHost: Could not find handler for "/SyncPlay/ListGroups"
[2020-07-03 20:02:12.451 +02:00] [ERR] [65] Emby.Server.Implementations.HttpServer.HttpListenerHost: Error processing request: "Unable to find the specified file". URL: "http://172.31.0.48:10006/SyncPlay/ListGroups"
```

Route for /SyncPlay/ListGroups returns 
```
Error processing request.
```

**Comment by sachk** [comment]
There seems to be a few places that've become mismatched in terms of wheter they're using GET or POST. I'd submit a pull request, but I'm not sure what approach is being taken for these choices so I'll leave that up to the maintainers.

There's a mismatch within SyncPlayService.cs however.

In the class the method is defined as GET
    public class SyncPlayListGroups : IReturnVoid
    {
        /// <summary>
        /// Gets or sets the filter item id.
        /// </summary>
        /// <value>The filter item id.</value>
        [ApiMember(Name = "FilterItemId", Description = "Filter by item id", IsRequired = false, DataType = "string", ParameterType = "query", Verb = "GET")]
        public string FilterItemId { get; set; }
    }

Over here, it's still defined as a POST.
`public List<GroupInfoView> Post(SyncPlayListGroups request)`

**Comment by cvium** [comment]
The name of the function is 100% irrelevant. The route and HTTP verb is defined in the `SyncPlayListGroups` class. The problem is that this line wasn't updated in the api client PR https://github.com/jellyfin/jellyfin-apiclient-javascript/pull/54/files#diff-05dbc2d65996d158626249e0c9f2d051R3272

**Comment by MrTimscampi** [comment]
@cvium It is actually updated in 1.3.1: https://github.com/jellyfin/jellyfin-apiclient-javascript/blob/master/src/apiClient.js#L3270

But the issue is that the JS apiclient has only one function for all SyncPlay endpoints, which is a pretty bad design.

**Comment by cvium** [comment]
So all the POST requests are broken then? 🙄

**Comment by peperunas** [comment]
I'm having a similar issue but with a different endpoint. The issue might be the same.

The setup is very similar indeed, I'm using the latest docker `jellyfin/jellyfin:unstable` image.

```
[2020-07-10 21:35:03.653 +00:00] [ERR] [25] Emby.Server.Implementations.HttpServer.HttpListenerHost: Error processing request. URL: "http://example.com/SyncPlay/List"
System.NotImplementedException: Could not find method named Get(SyncPlayList) or Any(SyncPlayList) on Service SyncPlayService
```

**Comment by crobibero** [comment]
Fixed by #3535
